js/foundation.offcanvas.js

```diff
@@ -136,6 +137,20 @@ class OffCanvas extends Plugin {
       this.$element.css('transition-duration', this.options.transitionTime);
     }
 
+    let inCanvasFor = this.$element.attr('class').match(/\bin-canvas-for-(\w+)/);
+
+    if (!this.options.inCanvasOn && inCanvasFor && inCanvasFor.length === 2) {
+      // Extract breakpoint name from CSS class if no inCanvasOn option set but using .in-canvas-for-[BREAKPOINT]
+      this.options.inCanvasOn = inCanvasFor[1];
+    } else if (this.options.inCanvasOn && !inCanvasFor) {
+      // Add CSS class if only inCanvasOn option set
+      this.$element.addClass(`in-canvas-for-${this.options.inCanvasOn}`);
+    }
```

Comment:
What if:
```html
<div data-off-canvas data-in-canvas-for="medium" class="off-canvas in-canvas-for-large">
```

With that case, the breakpoint is considered as "medium" in Js but "large" in CSS. Here how I see things to ensure a consistent behavior:
* Classname is a (backward-compatible) way to set the option.
* The option also require require the class to be set to work properly.

And about priorities: component class (less common = more specific = prioritized) > component options > default options.

So according to me, the check should be the following (pseudo-code):
```js
// Set the `inCanvasOn` option if a class is found
if (inCanvasSplit.length === 2) {
  this.options.inCanvasOn = inCanvasSplit[1]
}

// Ensure the required class is set
if (this.options.inCanvasOn) {
  this.$element.addClass(`in-canvas-for-$(this.options.inCanvasOn}`);
}
```